wremnants/muon_selections.py

```diff
@@ -41,26 +43,23 @@ def select_good_muons(df, ptLow, ptHigh, datasetGroup, nMuons=1, use_trackerMuon
 
     return df
 
-def define_trigger_muons(df, what_analysis=ROOT.wrem.AnalysisType.Dilepton):
-
-    if what_analysis == ROOT.wrem.AnalysisType.Dilepton:
-        # by convention define trigMuons as the positive charge, but actually both leptons could be triggering here 
-        # TODO: rename to something more meaningful, like positive/negative, or first/second
-        df = df.DefinePerSample("trigMuons_charge0", "1")
-        df = df.DefinePerSample("nonTrigMuons_charge0", "-1")
-    elif what_analysis == ROOT.wrem.AnalysisType.Wlike:
-        # mu- for even event numbers, mu+ for odd event numbers
-        df = df.Define("trigMuons_charge0", "event % 2 == 0 ? -1 : 1")
-        df = df.Define("nonTrigMuons_charge0", "-trigMuons_charge0")
+def define_trigger_muons(df, name_first="trigMuons", name_second="nonTrigMuons", dilepton=False):
+    if dilepton:
+        # by convention define first as negative charge, but actually both leptons could be triggering here 
+        logger.debug(f"Using dilepton trigger selection, the negative (positive) muon collection is named {name_first} ({name_second})")
+        df = df.DefinePerSample(f"{name_first}_charge0", "-1")
+        df = df.DefinePerSample(f"{name_second}_charge0", "1")
     else:
-        # Wmass currently doesn't call this function
-        raise NotImplementedError(f"define_trigger_muons(): no implementation for analysis {what_analysis}")
+        # mu- for even event numbers, mu+ for odd event numbers        
+        logger.debug(f"Using w-like trigger selection, the trigger (non trigger) muon collection is named {name_first} ({name_second})")
+        df = df.Define(f"{name_first}_charge0", "event % 2 == 0 ? -1 : 1")
```

Comment:
Also here for the event parity, we should probably define a isEvenEvent or isOddEvent with DefinePerSample, it would be more immediate, in addition to saving a few operations

Reply:
I don't think saving operations makes any difference, but it could be more clear if we use it multiple times, indeed

Reply:
I think DefinePerSample only works for constant values, so I have to use Define.

Reply:
Yes sure, I wrote it wrong (too bad we don't have a definePerHalfSample :p )